### DISCUSSION TITLE
Adding colors to switch blocks.

### DISCUSSION THREAD

**Discussion by UdayakumarHidakal** [plantuml/plantuml#1054] [Q&A] [answered]
- Is there any way to add colors to the switch blocks?

**Comment by Strange21** [comment]
So we want to add color to switch block.
following is the code snippet we are using 
```
@startuml
start
switch (test?)
case ( condition A )
  :Text 1;
case ( condition B ) 
  :Text 2;
endswitch
stop
@enduml
```

**Comment by The-Lum** [comment, marked as answer]
Hello @UdayakumarHidakal, @Strange21, and all,

- What element would you like to color?

If it is the diamond of the switch's test; Here is a proposal:
```puml
@startuml
<style>
diamond  {
  BackgroundColor White/Green
}
</style>

start
switch (test?)
case ( condition A )
  :Text 1;
case ( condition B ) 
  :Text 2;
endswitch
stop

@enduml
```
>[![](https://www.planttext.com/api/plantuml/svg/PSwn3i8m30JGtLznOpDLCD8AIXdu0IJcAB5eH1kZn1KWnByJ0M9XzTqzkSjgayxJI6tM-yXhyi5E4Ztm8A2txdnACaU_ar4IZaDGhlU9ETAJslgxeRT2-HhK3QYKisuCEPiP5LpHWWQ9sC8KStNWcsBH_ESz37w5PKCS_OSilbo8kd8ehxu0)](https://www.planttext.com/?text=PSwn3i8m30JGtLznOpDLCD8AIXdu0IJcAB5eH1kZn1KWnByJ0M9XzTqzkSjgayxJI6tM-yXhyi5E4Ztm8A2txdnACaU_ar4IZaDGhlU9ETAJslgxeRT2-HhK3QYKisuCEPiP5LpHWWQ9sC8KStNWcsBH_ESz37w5PKCS_OSilbo8kd8ehxu0)

See also answer here:
- https://forum.plantuml.net/15691/how-to-change-color-single-switch-diamond-activity-diagram

Regards.

**Reply by UdayakumarHidakal** [reply]
Thank you @The-Lum ,

- This make sense, It works.

Regards,
@UdayakumarHidakal @Strange21
